internal/tests/api/nodeset_test.go

```diff
@@ -100,24 +89,21 @@ func TestNodeSetRegistration_Registered(t *testing.T) {
 }
 
 // Cleanup after a unit test
-func nodeset_cleanup(snapshotName string) {
+func nodeset_cleanup(snapshotName string, dependency func(*testing.T), t *testing.T) {
 	// Handle panics
 	r := recover()
 	if r != nil {
 		debug.PrintStack()
 		fail("Recovered from panic: %v", r)
 	}
-
 	// Revert to the snapshot taken at the start of the test
-	if snapshotName != "" {
-		err := testMgr.RevertToCustomSnapshot(snapshotName)
-		if err != nil {
-			fail("Error reverting to custom snapshot: %v", err)
-		}
+	err := testMgr.DependsOn(dependency, &snapshotName, t)
```

Comment:
We may as well hardcode the baseline snapshot here instead of making it a parameter, because if the intent is *each test always reverts to the baseline at the end, even if it restored a different one at the start*, which it seems like it's doing, then this would cut down on some bloat.